src/org/jetbrains/java/decompiler/modules/decompiler/IfHelper.java

```diff
@@ -158,7 +157,7 @@ private static boolean collapseIfIf(IfNode rtnode) {
               if (ifinner.hasAnySuccessor()) {
                 StatEdge edge = ifinner.getFirstSuccessor();
                 if (edge.closure == ifchild) {
-                  edge.closure = null;
+                  edge.closure = ifparent;
```

Comment:
Shouldn't we also need to do this with all the edges that have `ifchild` as closure?